velox/common/memory/MemoryPool.cpp

```diff
@@ -483,6 +484,8 @@ MemoryPoolImpl::~MemoryPoolImpl() {
   if (destructionCb_ != nullptr) {
     destructionCb_(this);
   }
+
+  delete reclaimer_.load(std::memory_order_relaxed);
 }
```

Comment:
@pedroerp 

In old code, we destroyed  `reclaimer_` field inside `MemoryPoolImpl` after "end" of `~MemoryPoolImpl`.
In new code in the end we destroy `reclaimer_` manually.

So the two code is kind of same.